.github/workflows/ci-ubuntu-20.04-lts-workflow.yml

```diff
@@ -0,0 +1,72 @@
+name: CI-Ubuntu-20.04-LTS
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  gcc-rust-build-test:
+    name: gcc & rust build & test
+    runs-on: ubuntu-20.04
+    env:
+      # Share build cache when building rust API and the example project
+      CARGO_TARGET_DIR: ${{ github.workspace }}/target
+      CARGO_BUILD_JOBS: 32
+      CC: gcc
+      CXX: g++
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Ensure Python dependencies
+        run: |
+          pip install torch~=1.13 --extra-index-url https://download.pytorch.org/whl/cpu &&\
+          pip install --user -r tools/python_api/requirements_dev.txt -f https://data.pyg.org/whl/torch-1.13.0+cpu.html
+      
+      - name: Ensure Node.js dependencies
+        run: npm install --include=dev
+        working-directory: tools/nodejs_api
+
+      - name: Test with coverage
+        run: CC=gcc CXX=g++ make lcov NUM_THREADS=32
```

Comment:
GitHub-hosted machines have much fewer cores, please check this or use `$(nproc)` to get core counts dynamically.